lightwood/mixers/helpers/plinear.py

```diff
@@ -0,0 +1,171 @@
+            init.uniform_(self.bias, -bound, bound)
+
+    def w_discrete_normal(self):
+        """
+        Sample a w matrix based on a discrete normal distribution
+        :return: w
+        """
+
+        bucket_seed = random.uniform(0,1) # gett a number that can help determines which bucket of the distribution can numbers fall in
+        sigma_multiplier = 1 # this tell it how wide do we stretch the random saples
+
+        # set the widths for sampling, based on the seed
+        if bucket_seed > 0.33:
+            sigma_multiplier = 2
+            if bucket_seed > 0.8:
+                sigma_multiplier = 4
+
+        # generate the initial tensor, this will ultimately transforms in to the weights
+        w = torch.Tensor(self.out_features, self.in_features)
+        # make sure that they are evently distributed between -1, 1
+        init.uniform_(w, a=-1, b=1)
+
+        # adjust based on sigma
+        w = self.mean*( 1+ w * torch.abs(self.sigma) * sigma_multiplier )
+        # you can see how the average sigma changes over trainings
+        #print(torch.mean(self.sigma))
+        return w
+
+
+    def w_normal(self):
+        """
+        This samples a w based on the distribution parameter matrices self.mean, self.sigma
+        @:return w
+        """
+
+        # for this we generate matrices for means and sigmas in distributions on the Probablility axis of the probability distribution for the weights
+        mu_y = 1 / (torch.abs((self.sigma ** 2) * math.pi * 2)) ** (0.5)
+        sigma_y = torch.sqrt(torch.abs(mu_y - mu_y * (math.e ** (-0.5))))
+        mu_y = mu_y.tolist()
+        sigma_y = sigma_y.tolist()
+
+        # now we can generate samples
+
+        y_rand = [[float(np.random.normal(mu_y[y_i][x_i], sigma_y[y_i][x_i], 1)[0]) for x_i, x in enumerate(y)] for
+                  y_i, y in enumerate(sigma_y)]
+        y_rand = torch.Tensor([[y_rand[y_i][x_i] if y_rand[y_i][x_i] <= mu_y[y_i][x_i] else y_rand[y_i][x_i] -
+                                                                                            mu_y[y_i][x_i] for x_i, x in
+                                enumerate(y)] for y_i, y in enumerate(sigma_y)])
+
+        # print(y_rand)
+
+        x_sign = torch.Tensor(
+            [[1 if random.randint(0, 1) > 0 else -1 for x_i, x in enumerate(y)] for y_i, y in enumerate(sigma_y)])
+
+        # now we bind the sigmas and means to the graph so they can be aligned in backpropagation
+        w = self.mean + torch.sqrt(torch.abs(
+            -2 * (self.sigma ** 2) * torch.log(torch.abs(y_rand * torch.sqrt(2 * np.pi * (self.sigma ** 2)))))) * x_sign
+
+        return w
+
+    def forward(self, input):
+        '''
+        Forward pass of the function.
+        The goal is to generate values such as if they weights of the linear operation are sampled from a normal distribution
+        '''
+
+        return F.linear(input, self.w_sampler(), self.bias)
+
+
+if __name__ == "__main__":
```

Comment:
I assume this was an old test, maybe replace with a test for the PLinear class itself instead :?